bindings/netframework/ElectionGuard.NetFramework/ElectionGuardInterop/Ballot.h

```diff
@@ -179,6 +259,156 @@ namespace ElectionGuardInterop
+              std::make_unique<electionguard::ElementModQ>(*cryptoHash->_instance),
+              std::make_unique<electionguard::ConstantChaumPedersenProof>(*proof->_instance)
+            );
+        }
+        internal
+            : CiphertextBallotContest(std::unique_ptr<electionguard::CiphertextBallotContest> other)
+        {
+            this->_instance = other.release();
+        }
+        CiphertextBallotContest(electionguard::CiphertextBallotContest *unowned)
+            : ManagedInstance(unowned, false)
+        {
+        }
+        CiphertextBallotContest(const electionguard::CiphertextBallotContest *unowned)
+            : ManagedInstance(unowned)
+        {
+        }
+
+      public:
+        property String ^
+          ObjectId {
+              String ^ get() {
+                  auto unmanaged = this->_instance->getObjectId();
+                  return gcnew String(unmanaged.c_str());
+              }
+          }
+
+          property ElementModQ ^
+          DescriptionHash {
+              ElementModQ ^ get() {
+                  auto unmanaged = this->_instance->getDescriptionHash();
+                  return gcnew ElementModQ(unmanaged);
+              }
+          }
+
+          property array<CiphertextBallotSelection ^> ^
+          Selections {
+              array<CiphertextBallotSelection ^> ^ get() {
+                  auto unmanaged = this->_instance->getSelections();
+                  auto elements = gcnew array<CiphertextBallotSelection ^>(unmanaged.size());
+
+                  int index = 0;
+                  for (const auto &item : unmanaged) {
+                      elements[index] = gcnew CiphertextBallotSelection(&item.get());
+                      index++;
+                  }
+
+                  return elements;
+              }
+          }
+
+          property ElementModQ ^
+          Nonce {
+              ElementModQ ^ get() {
+                  auto unmanaged = this->_instance->getNonce();
+                  return gcnew ElementModQ(unmanaged);
+              }
+          }
+
+          property ElGamalCiphertext ^
+          CiphertextAccumulation {
+              ElGamalCiphertext ^ get() {
+                  auto unmanaged = this->_instance->getCiphertextAccumulation();
+                  return gcnew ElGamalCiphertext(unmanaged);
+              }
+          }
+
+          property ElementModQ ^
+          CryptoHash {
+              ElementModQ ^ get() {
+                  auto unmanaged = this->_instance->getCryptoHash();
+                  return gcnew ElementModQ(unmanaged);
+              }
+          }
+
+          property ConstantChaumPedersenProof ^
+          Proof {
+              ConstantChaumPedersenProof ^ get() {
+                  auto unmanaged = this->_instance->getProof();
+                  return gcnew ConstantChaumPedersenProof(unmanaged);
+              }
+          }
+
+          ElementModQ ^
+          aggregateNonce() {
```

Comment:
this is an example of where the property returned is actually owned by the caller.  Since we are in managed land, that is transparent to C# since we move the unique pointer into the ElementModQ managed wrapper.  The GC handles disposing the object automatically.